src/pages/PassportRedemption/SignIn/uploadScreen.tsx

```diff
@@ -0,0 +1,302 @@
+import React, { useState, useEffect } from 'react';
+import { useParams, useHistory } from 'react-router-dom';
+import { useTranslation } from 'react-i18next';
+import styled from 'styled-components';
+
+import { SubTitle } from '../style';
+import { InputContainer, Label, Column, InputField } from './index';
+import { Button } from './emailScreen';
+import {
+  getUploadUrl,
+  sendImage,
+  getAllParticipatingSellers,
+  uploadCrawlReceipts,
+} from '../../../utilities/api/interactionManager';
+import uploadImageIcon from '../Assets/uploadImage.png';
+
+const UploadScreen = () => {
+  const { push } = useHistory();
+  const { t } = useTranslation();
+  const { id } = useParams();
+
+  // all sellers part of the lny crawl - retrieved on initial load of the page
+  const [allParticipatingSellers, setallParticipatingSellers] = useState<any>(
+    []
+  );
+
+  // seller name + id
+  const [participatingSeller, setParticipatingSeller] = useState('');
+  const [participatingSellerId, setParticipatingSellerId] = useState<number>(
+    -1
+  );
+
+  // search values + did user select a valid merchant?
+  const [sellersSearchBar, setSellersSearchBar] = useState<any>([]);
+  const [selectedSeller, setSelectedSeller] = useState(true);
+
+  // receipt total
+  const [billTotal, setBillTotal] = useState('');
+
+  // string value of receipt + actual receipt (file format)
+  const [receiptFilePath, setReceiptFilePath] = useState('');
+  const [receipt, setReceipt] = useState<File | null>(null);
+
+  useEffect(() => {
+    const getAllSellers = async () => {
+      const { data: allSellers } = await getAllParticipatingSellers();
+      setallParticipatingSellers(allSellers);
+    };
+    getAllSellers();
+  }, []);
+
+  useEffect(() => {
+    const debouncer = setTimeout(() => {
+      if (allParticipatingSellers[0]?.name) {
+        let match = allParticipatingSellers
+          .filter((merchant) =>
+            merchant?.name
+              .toLowerCase()
+              .includes(participatingSeller.toLowerCase().trim())
+          )
+          .slice(0, 5);
+        setSellersSearchBar(match);
+      }
+    }, 200);
+    return () => clearTimeout(debouncer);
+  }, [participatingSeller, allParticipatingSellers]);
+
+  const uploadImage = async (event) => {
+    event.preventDefault();
+    if (event.target.files && event.target.files[0]) {
+      setReceiptFilePath(URL.createObjectURL(event.target.files[0]));
+      setReceipt(event.target.files[0]);
+    }
+  };
+
+  const submitReceipt = async () => {
+    if (receipt === null) return false;
+
+    try {
+      const ext = receipt.type.split('/')[1];
+
+      let filename = `${id}-${new Date()
+        .toISOString()
+        .replace(/(:|\.)/g, '-')}.${ext}-${participatingSeller}`;
```

Comment:
could we actually do - id, then participating seller, then date last? might just be easier when we're looking these up later

Reply:
yep! updated